static/js/controllers/contacts.js

```diff
@@ -240,18 +236,25 @@ var _ = require('underscore'),
           $scope.filtered = true;
           liveList = LiveList['contact-search'];
           liveList.set([]);
-          _query();
+          return _query();
```

Comment:
Interestingly I had to do this to get tests to pass, and I'm not sure how they were passing without this.

A lot of the tests of the contacts controller wait for the `setupPromise` to resolve and then check things, but the setupPromise chain was being broken here, meaning that in a normal system `setupPromise` would resolve before searches had been complete. This should have directly caused problems, but didn't until I added the `$q.all` below. I suspect we were either very lucky or our unit test `Q` implementation is broken.